### DISCUSSION TITLE
web：使用setComposition实现多段pag动画的衔接，在手机上放久了会报错，安卓表现为动画卡住不播，ios动画会消失

### DISCUSSION THREAD

**Discussion by ligeisme** [Tencent/libpag#1912] [Q&A]
DOMException，code：20，INDEX_SIZE_ERR:1，DOMSTRING_SIZE_ERR:2

  // 比如 有1,2两个动画，交替播放，一段时间后再就会报错
    const PAG = await window.libpag.PAGInit();
    const buffer1 = await fetch('../assets/jump.pag').then((response) => response.arrayBuffer());
    const pagFile1 = await PAG.PAGFile.load(buffer1);

    const buffer2 = await fetch('../assets/hello.pag').then((response) => response.arrayBuffer());
    const pagFile2 = await PAG.PAGFile.load(buffer2);

    const canvas = document.getElementById('pag');
    canvas.width = 500;
    canvas.height = 500;

    let pagView = await PAG.PAGView.init(pagFile1, canvas);
    pagView.addListener('onAnimationEnd', async () => {
        start(pagView);
    });
    async function start() {
      v = v === 1 ? 2 : 1;
      pagView.setComposition(v === 2 ? pagFile2 : pagFile1)
      pagView?.setRepeatCount(1);
      pagView?.play();
    }
    start();

**Comment by whatwg6** [comment]
切后台了？？监听切后台事件，前台展示调用 play，后台调 pause

**Reply by ligeisme** [reply]
好的，我试试看，不过偶尔也出现过放着一段时间卡死了，切后台再回来消失的情况，不过概率不高

**Reply by ligeisme** [reply]
还是解决不了，有一个DOM报错，code：20，INDEX_SIZE_ERR:1，DOMSTRING_SIZE_ERR

**Comment by zhalice2011** [comment]
iPhone 是处于省电模式吗

**Reply by ligeisme** [reply]
不是，只要放久了就有概率报错

**Reply by ligeisme** [reply]
试了下是必现的，看了下报错有一个DOM报错，code：20，INDEX_SIZE_ERR:1，DOMSTRING_SIZE_ERR
我改成页面切换到前台的时候销毁重建pagView依然解决不了，基本上通过setComposition交替播放素材半小时就会出这个索引超出的报错，然后动画就无了

**Comment by 2574161467** [comment]
同样碰到了动画消失的情况 ios会消失 也没看到报错 重启手机就好了 请问是什么原因呢

**Reply by 2574161467** [reply]
ios省电模式会有报错 这种非省电模式也没有报错信息 看打印也是已经创建完成 就是不显示 调用play stop等方法也没有报错信息

**Comment by CodeJhF** [comment]
感谢 @2574161467 的回复，也抱歉我们没有及时跟进这条。

如果问题仍存在，请在最新版本（4.4 / 4.5）上复现并补充版本号与 demo，我们会优先跟进，本条先关闭。（stale-close）
